crates/core/tedge_api/src/entity_store.rs

```diff
@@ -216,19 +231,41 @@ impl EntityStore {
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct EntityMetadata {
-    parent: Option<EntityId>,
-    r#type: EntityType,
-    external_id: String,
-    other: serde_json::Value,
+    pub parent: Option<EntityId>,
+    pub r#type: EntityType,
+    pub external_id: String,
+    pub other: serde_json::Value,
 }
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-enum EntityType {
+pub enum EntityType {
     MainDevice,
     ChildDevice,
     Service,
 }
 
+impl EntityMetadata {
+    /// Creates a entity metadata for a child device.
+    pub fn main_device(external_id: String) -> Self {
+        Self {
+            external_id,
+            r#type: EntityType::MainDevice,
+            parent: None,
+            other: serde_json::json!({}),
+        }
+    }
+
+    /// Creates a entity metadata for a child device.
+    pub fn child_device(external_id: String) -> Self {
+        Self {
+            external_id,
+            r#type: EntityType::ChildDevice,
+            parent: Some("device/main//".to_string()),
+            other: serde_json::json!({}),
+        }
+    }
+}
+
```

Comment:
> I see this functions are only used in tests

Indeed, currently only used for testing. Mostly because creating an `EntityRegistrationMessage` where a bit more painful.

>  Would you say that we'll need to construct `EntityMetadata` in normal code as well? My intent was for only `EntityStore` to be able to construct it, when parsing a registration message, but this can change. 

Not sure to see the benefits of  `EntityRegistrationMessage`.